src/DependencyInjection/MongoDBExtension.php

```diff
@@ -61,16 +62,14 @@ public static function createClientServiceId(string $clientId): string
 
     private function createClients(string $defaultClient, array $clients, ContainerBuilder $container): void
     {
-        $clientPrototype = $container->getDefinition('mongodb.prototype.client');
-
         foreach ($clients as $client => $configuration) {
             $serviceId = self::createClientServiceId($client);
 
-            $clientDefinition = clone $clientPrototype;
+            $clientDefinition = new ChildDefinition('mongodb.abstract.client');
             $clientDefinition->setArgument('$uri', $configuration['uri']);
             $clientDefinition->setArgument('$uriOptions', $configuration['uriOptions'] ?? []);
             $clientDefinition->setArgument('$driverOptions', $configuration['driverOptions'] ?? []);
-
+            $clientDefinition->addTag('mongodb.client', ['name' => $client]);
```

Comment:
What about `mongodb.client.name`?

Reply:
The tag name `mongodb.client` seems correct. I don't understand your proposition.

Reply:
Ah sorry, yes the tag is correct and uses the name of the options. All good, sorry